File: Jan/Week 2/LeetCode/problem_3.cpp
//  Day 10 Problem No 20

// 238. Product of Array Except Self => https://leetcode.com/problems/product-of-array-except-self/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n,1);
        int l = 1;
        int r = 1;
        for(int i=0;i<n;i++){
            ans[i] = l;
            l = l * nums[i];
        }
        for(int i=n-1;i>=0;i--){
            ans[i] = ans[i] * r;
            r = r * nums[i];
        }
        return ans;
    }
};

File: Jan/Week 3/LeetCode/problem_3.py
# Day 17 Problem No 37

# 392. Is Subsequence => https://leetcode.com/problems/is-subsequence/description/?envType=study-plan-v2&envId=leetcode-75

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0
        j = 0
        while i < len(s) and j < len(t):
            if s[i] == t[j]:
                i+=1
            j+=1

        return i == len(s)